apptail/storage/tail_tracker.go

```diff
@@ -0,0 +1,231 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/ActiveState/log"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Tracker interface {
+	Update(instKey string, childKey string, childVal int64)
+	LoadTailers()
+	Remove(key string)
+	RegisterInstance(instKey string)
+	InitializeChildNode(instKey string, childkey string, offSet int64)
+	Commit() error
+	StartSubmissionTimer(persistInterval time.Duration)
+	IsInstanceRegistered(instKey string) bool
+	IsChildNodeInitialized(instKey string, childkey string) bool
+	GetFileCachedOffset(instkey string, fname string) int64
+	getBuffer() ([]byte, error)
+	CleanUp(validIds map[string]bool)
+}
+
+type BoxedInt64 struct{ V int64 }
+
+type TailNode map[string]*BoxedInt64
+
+type Tailer struct {
+	Instances map[string]TailNode
+}
+
+type tracker struct {
+	storage       Storage
+	Cached        *Tailer // do not expose this, it should ONLY be updated via Tracker methods
+	mux           *sync.Mutex
+	writeMux      *sync.Mutex
+	timerStopChan chan struct{} // used to send quit signal to time,
+	debug         bool
+}
+
+func NewTracker(s Storage, debug bool) Tracker {
+
+	return &tracker{
+		storage:  s,
+		mux:      &sync.Mutex{},
+		writeMux: &sync.Mutex{},
```

Comment:
I feel like `writeMux` should live on the `FileStorage` instead and be delt with inside `FileStorage#Write` (since that's responsible for the storageness of things); thoughts?